salt/_pillar/metalk8s.py

```diff
@@ -15,8 +15,20 @@ def _load_config(path):
     log.debug('Loading MetalK8s configuration from %s', path)
 
     config = None
-    with salt.utils.files.fopen(path, 'rb') as fd:
-        config = salt.utils.yaml.safe_load(fd)
+    try:
+        with salt.utils.files.fopen(path, 'rb') as fd:
+            config = salt.utils.yaml.safe_load(fd) or {}
+    except Exception as exc:
+        return __utils__['pillar_utils.errors_to_dict']([
+            "Failed to load {}: {}".format(path, exc)
+        ])
+    if not config:
+        error_tplt = (
+            'Invalid BootstrapConfiguration at {}: must not be empty'
```

Comment:
More than that, must be a valid config, so I'd generalize the error message.

Reply:
No, here we **know** the config is empty. We'll have better messages later in the validation process.

Reply:
Playing devil's advocate:

The error reporting tells me something is wrong ("must not be empty"). It doens't tell me to enter a valid `BootstrapConfiguration` or whatnot. So playing the 'dumb' card, I could put `blah` in that file, and then I should've fixed the issue, right?

Nope, re-running bootstrap will now give me *another* error, telling me to put a valid `BootstrapConfiguration` in that file. Why didn't it tell me to do so from the beginning?

Reply:
@gdemonet why the thumbs down??

If we reached this state, it means we loaded an empty `dict`(or user created an empty Bootstrap file)

Maybe we print directly that  `Error: Empty Bootstrap configuration` ??

Reply:
Ohhhh right! My bad, didn't think of this, I'm actually used to error messages that are useful for the dev/support (i.e. useful to pinpoint the source of the issue), but forgot about the whole "user is discovering this new product" story. Good point! We could add a common "hint" / "usage" message in the Bootstrap script in case `'_errors' in pillar.metalk8s`.